pkg/miscellaneous/miscellaneous.go

```diff
@@ -349,3 +355,52 @@ func SleepWithContext(ctx context.Context, sleepFor time.Duration) error {
 		}
 	}
 }
+
+// IsBackupBucketEmpty checks whether the backup bucket is empty or not.
+func IsBackupBucketEmpty(snapStoreConfig *brtypes.SnapstoreConfig, logger *logrus.Logger) bool {
+	logger.Info("Checking whether the backup bucket is empty or not...")
+
+	if snapStoreConfig == nil || len(snapStoreConfig.Provider) == 0 {
+		logger.Info("storage provider name not specified")
+		return true
+	}
+	store, err := snapstore.GetSnapstore(snapStoreConfig)
+	if err != nil {
+		logger.Fatalf("failed to create snapstore from configured storage provider: %v", err)
+	}
+
+	baseSnap, deltaSnapList, err := GetLatestFullSnapshotAndDeltaSnapList(store)
+	if err != nil {
+		logger.Errorf("failed to list the snapshot: %v", err)
+		return false
+	}
+
+	if baseSnap == nil && (deltaSnapList == nil || len(deltaSnapList) == 0) {
+		logger.Infof("No snapshot found. BackupBucket is empty")
+		return true
+	}
+	return false
+}
+
+// GetInitialClusterState returns the cluster state, either `new` or `existing`.
+func GetInitialClusterState(ctx context.Context, logger logrus.Entry, clientSet client.Client, podName string, podNS string) string {
```

Comment:
Can you please use the constants `ClusterStateNew` and `ClusterStateExisting` that you created instead of hardcoding?

Reply:
done. PTAL. Also added one more unit tests